packages/terra-functional-testing/src/terra-cli/wdio/test-runner.js

```diff
@@ -29,19 +32,21 @@ class TestRunner {
    * @returns {Promise} A promise that resolves with the test run exit code.
    */
   static async run(options) {
+    let exitCode;
     try {
       const { config } = options;
 
       const configPath = TestRunner.configPath(config);
       const testRunner = new Launcher(configPath);
 
-      const exitCode = await testRunner.run();
-
-      process.exit(exitCode);
+      exitCode = await testRunner.run();
     } catch (error) {
-      // eslint-disable-next-line no-console
-      console.error('[terra-functional-testing] Launcher failed to start the test.\n', error);
-      process.exit(1);
+      logger.error('Launcher failed to start the test');
```

Comment:
Should this set the exit code to 1 still?

Reply:
From what I've read, using process.exit is strongly discouraged as it immediately kills the process.  Sometimes things get buffered like stdout or other asynchronous tasks and if you process.exit it will end them prematurely. Throwing the error will pass up to the command runner and ultimately end the process with a non success status, so I think this should work unless there was a reason we were hard exiting I'm not aware of.

Reply:
I think throwing the error is fine here instead of process.exit. For historical context the process.exit is the documented approach to using the test runner programmatically:

https://webdriver.io/docs/clioptions.html#run-the-test-runner-programmatically

It's also how the default runner is setup:

https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-cli/src/commands/run.js#L120-L136